### PR TITLE
Adds Pre-Commit Hooks, Updates Requirements.txt &  Extends .gitignore

### PR DESCRIPTION
**Pull request by helmigsimon**
### Background
I noticed that there are some minor issues in the repository:
- Not all dependencies are tagged to a specific version number -> This can lead to code breaking unexpectedly
- There is no automation of code formatting & validity checks -> This leads to inconsistency in the codebase and contributes/maintainers having to deal with unnecessary categories of bugs (i.e. AST violations)
- The `.gitignore` is rather short and doesn't include general Python settings outside of project specific exclusions

### Changes
To address the issues I identified, I have made the folloiwng changes:
- Added pre-commit hooks to the repository, such that certain classes of checks can be executed automatically 
- Tagging packages with the versions downloaded by default as per 2023/04/07
- Introducing dev-requirements.txt for dev packages (only including `pre-commit` for now)
- Deleting ai_settings.yaml as it is unused and in .gitignore

### Test Plan

As these are not functional changes, I have not tested particular cases other than running the scripts/main.py file. 
The script continues to run as expected.

### Change Safety

- [ ] I have added tests to cover my changes
- [ ] I have considered potential risks and mitigations for my changes

### CODE REVIEW

File: ai_settings.yaml
Comment (nponeccop) [inline ai_settings.yaml:1]: This change shouldn't be in the PR, should it?

File: requirements.txt [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+gTTS==2.3.1
 openai==0.27.2
+pinecone-client==2.2.1
 playsound==1.2.2
 python-dotenv==1.0.0
 pyyaml==6.0
 readability-lxml==0.8.1
-requests
+requests==2.28.2
+requests==2.28.2
```
Comment (nponeccop) [inline requirements.txt:14]: Requests lines are duplicated

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: The giant .gitignore and pre-commit hooks should be separate PRs.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Splitting the PR and reducing the scope of individual changes (e.g. just add the versions without sorting the dependencies) would be nice.

**Comment by nponeccop** [comment]
@helmigsimon Now it's additionally out of date. Resolve the conflicts.

**Comment by richbeales** [comment]
closed as resolved by #1393

**Pull request closed without merging**